python/guide/module/autograd.md

```diff
@@ -1,178 +1,260 @@
+
+## How do we calculate gradients?
+
+We differentiate. MXNet Gluon uses Reverse Mode Automatic Differentiation (`autograd`) to backpropagate gradients from the loss metric to the network parameters.
+
+<p align="center">
+    <img src="./imgs/autograd_forward_backward.png" alt="drawing" width="400"/>
+</p>
+
+Long Answer:
+
+One option would be to get out our calculus books and work out the gradients by hand. Who wants to do this though? It's time consuming and error prone for starters. Another option is Symbolic Differentiation, which calculates the formulas for each gradient, but this quickly leads to incredibly long formulas as networks get deeper and operators get more complex. We could use finite differencing, and try slight differences on each parameter and see how the loss metric responds, but this is computationally expensive and has poor numerical precision.
+
+What's the solution? Use automatic differentiation to backpropagate the gradients from the loss metric back to each of the parameters. Sometimes this is called reverse mode automatic differentiation, and it's very efficient in 'fan-in' situations where many parameters effect a single loss metric. Although forward mode automatic differentiation methods exist, they're suited to 'fan-out' situations where few parameters effect many metrics, which isn't the case for training neural networks.
+
+## How does Automatic Differentiation (`autograd`) work?
+
+Stage 1. Create a record of the operators used by the network to make predictions and calculate the loss metric. Called the 'forward pass' of training.
+Stage 2. Work backwards through this record and evaluate the partial derivatives of each operator, all the way back to the network parameters. Called the 'backward pass' of training.
+
+<p align="center">
+    <video width="400" controls playsinline autoplay muted loop>
+        <source src="./imgs/autograd_graph.mp4" type="video/mp4">
+    </video>
+</p>
+
+Long Answer:
+
+All operators in MXNet have two methods defined: a `forward` method for executing the operator as expected, and a `backward` method that returns the partial derivative (the derivative of the output with respect to the input). On the vary rare occasion you need to implement your own custom operator, you'll define the same two methods.
+
+Automatic Differentiation creates a record of the operators used (i.e. the `forward` method calls) by the network to make predictions and calculate the loss metric. A graph structure is used to record this, capturing the inputs (including their value) and outputs for each operator and how the operators are related. We call this the 'forward pass' of training.
```

Comment:
no need for capitalization (we can talk offline about an alternative 'emphasis' to use)

Reply:
done.